vars/neoDeploy.groovy

```diff
@@ -8,222 +11,218 @@ import com.sap.piper.tools.ToolDescriptor
-            }
-        } else if(deployMode == 'mta') {
-            warAction = 'deploy-mta'
-        }
+            def warAction = utils.getParameterInValueRange(script, configuration, 'warAction', warActions)
 
-        if (deployMode == 'warPropertiesFile') {
-            propertiesFile = utils.getMandatoryParameter(configuration, 'propertiesFile')
-            if (!fileExists(propertiesFile)){
-                error "Properties file cannot be found with parameter propertiesFile: '${propertiesFile}'."
+            if (warAction == 'rolling-update') {
+                if (!isAppRunning(neoCommandHelper)) {
+                    warAction = 'deploy'
+                    echo "Rolling update not possible because application is not running. Falling back to standard deployment."
+                }
             }
-        }
 
-        if (deployMode == 'warParams') {
-            applicationName = utils.getMandatoryParameter(configuration, 'applicationName')
-            runtime = utils.getMandatoryParameter(configuration, 'runtime')
-            runtimeVersion = utils.getMandatoryParameter(configuration, 'runtimeVersion')
-            def vmSizes = ['lite', 'pro', 'prem', 'prem-plus']
-            vmSize = configuration.vmSize
-            if (! (vmSize in vmSizes)) {
-                throw new Exception("[neoDeploy] Invalid vmSize = '${vmSize}'. Valid 'vmSize' values are: ${vmSizes}.")
+            echo "Link to the application dashboard: ${neoCommandHelper.cloudCockpitLink()}"
+
+            if (warAction == 'rolling-update') {
+                sh neoCommandHelper.rollingUpdateCommand()
+            } else {
+                sh neoCommandHelper.deployCommand()
+                sh neoCommandHelper.restartCommand()
             }
-        }
 
-        if (deployMode in ['mta','warParams']) {
-            deployHost = utils.getMandatoryParameter(configuration, 'host')
-            deployAccount = utils.getMandatoryParameter(configuration, 'account')
-        }
 
-        def neo = new ToolDescriptor('SAP Cloud Platform Console Client', 'NEO_HOME', 'neoHome', '/tools/', 'neo.sh', null, 'version')
-        def neoExecutable = neo.getToolExecutable(this, configuration)
-        def neoDeployScript = """#!/bin/bash
-                                 "${neoExecutable}" ${warAction} \
-                                 --source "${archivePath}" \
-                              """
-
-        if (deployMode in ['mta', 'warParams']) {
-            neoDeployScript +=
-                    """--host '${deployHost}' \
-                    --account '${deployAccount}' \
-                    """
-        }
+        } else if (deployMode == 'mta') {
+            warAction = 'deploy-mta'
 
-        if (deployMode == 'mta') {
-            neoDeployScript += "--synchronous"
+            sh neoCommandHelper.deployMta()
         }
+    }
+    catch (Exception ex) {
+        echo "Error while deploying to SAP Cloud Platform. Here are the neo.sh logs:"
```

Comment:
Having the log is cool.

But: I would expect the toolset installed e.g to something like `/opt/sap/neo`. Under that path the user running Jenkins does normally not have write access to. Hence the log cannot be located here. For that purpose the neo toolset provides the variable `neo_logging_location`. With that it is possible to install the toolset to a location without write access, but to have the log at the expected location which is somewhere under /`var/log`, e.g. `/var/log/neo`.

We should at least be prepared to fetch the log from such another location. Additionally I think our neo docker image should have such a setup, but that is a different story.